remote-trait-object/src/ipc/multiplex.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::sync::mpsc::{channel, Receiver};
+use std::thread;
+
+pub struct MultiplexResult {
+    pub request_recv: Receiver<String>,
+    pub response_recv: Receiver<String>,
+    pub multiplexer: Multiplexer,
+}
+
+pub struct Multiplexer {
+    _receiver_thread: Option<thread::JoinHandle<()>>,
+}
+
+impl Multiplexer {
+    pub fn multiplex(ipc_recv: Receiver<String>) -> MultiplexResult {
+        let (request_send, request_recv) = channel();
+        let (response_send, response_recv) = channel();
+
+        let receiver_thread = thread::Builder::new()
+            .name("multiplexer".into())
+            .spawn(move || loop {
+                let original_message = ipc_recv.recv().unwrap();
+                let message = parse(original_message.clone());
```

Comment:
`parse` is a protocol. Please make this pluggable.

Reply:
Good. I'll add a "FIXME" comment.